### PR TITLE
msc2432 update

### PR DESCRIPTION
**Pull request by ram-nad**

### CODE REVIEW

File: lib/connection.cpp [hunk 1]
```diff
@@ -901,17 +902,11 @@ Room* Connection::roomByAlias(const QString& roomAlias, JoinStates states) const
...
 {
-    auto& aliasMap = d->roomAliasMap[aliasServer]; // Allocate if necessary
-    for (const auto& a : previousRoomAliases)
-        if (aliasMap.remove(a) == 0)
-            qCWarning(MAIN) << "Alias" << a << "is not found (already deleted?)";
-
-    for (const auto& a : roomAliases) {
-        auto& mappedId = aliasMap[a];
+    for (const auto& a: roomAliases) {
+        auto& mappedId = d->roomAliasMap[a];
```
Comment (KitsuneRal) [inline lib/connection.cpp]: You seem to have dropped an important part here that removes old aliases from the list. The way you have your code now, if some alias from `d->roomAliasMap` is in `previousRoomAliases` but not in `roomAliases` it seems to get stuck in `d->roomAliasMap` forever.

File: lib/connection.h [hunk 2]
```diff
@@ -141,8 +141,9 @@ class Connection : public QObject {
      *         it's an Invite rather than Join - to room pointers
      * \sa allRooms, rooms, roomsWithTag
      */
-    [[deprecated("Use allRooms(), roomsWithTag() or rooms(joinStates) instead")]]
-    QHash<QPair<QString, bool>, Room*> roomMap() const;
+    [[deprecated("Use allRooms(), roomsWithTag() or rooms(joinStates) "
+                 "instead")]] QHash<QPair<QString, bool>, Room*>
+    roomMap() const;
```
Comment (KitsuneRal) [inline lib/connection.h]: Clang-format is very bad with `[[attributes]]`, which is why I would advise against automatic formatting. This signature, in particular, should certainly be left intact.

File: lib/events/simplestateevents.h [hunk 3]
```diff
@@ -46,46 +48,130 @@ namespace EventContent {
...
+        }
+
+    public:
+        T1 value1;
+        T2 value2;
+
+    protected:
+        QString key1;
+        QString key2;
+    };
```
Comment (KitsuneRal) [inline lib/events/simplestateevents.h]: I'm not quite sure it's the best approach here - there are pretty few (standard) event types with exactly two items of content, and in such case I'd probably make a more basic non-template structure for the content. In any case, this doesn't belong to this header file anymore - please separate it (and `RoomCanonicalAliasEvent` definition) into another file.

File: lib/room.cpp [hunk 4]
```diff
@@ -2368,14 +2368,39 @@ Room::Changes Room::processStateEvent(const RoomEvent& e)
...
+
+            const auto previousAliases = std::move(previousAltAliases);
+
+            auto newAliases = cae.alt_aliases();
+
+            if (!cae.alias().isEmpty()) {
+                newAliases.push_front(cae.alias());
+            }
+
+            connection()->updateRoomAliases(id(), previousAliases, newAliases);
```
Comment (KitsuneRal) [inline lib/room.cpp]: You seem to have forgotten to incorporate the original code reacting to the event (the `setObjectName` line).

**Comment by ram-nad** [comment]
Thanks for the review. I will make the suggested changes. The clang-format seems to have formatted the files and I did use .clang-format provided with the file. I will update the PR without auto-formatting on.

**Pull request closed without merging**